lib/query_helper.rb

```diff
@@ -280,8 +280,12 @@ def search_filter(column_maps)
       raise ArgumentError.new("search_fields not defined") unless @search_fields.length > 0
       placement = :where
       maps = column_maps.select do |cm|
-        placement = :having if cm.aggregate
-        @search_fields.include? cm.alias_name
+        if @search_fields.include? cm.alias_name
+          placement = :having if cm.aggregate
```

Comment:
Does this placement have anything to do with the above condition ?

Reply:
yes, it do have. Earlier having was being applied in all the scenarios if search_fields is present even if we pass aggregate as false.